### PR TITLE
[ui] Enable filtering asset table by owner

### PR DESCRIPTION
**Pull request by clairelin135**
As the title.

Used in https://github.com/dagster-io/dagster/pull/20372

image

image

### CODE REVIEW

**Comment by clairelin135** [comment]
* **#20420** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20420?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clairelin135 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by salazarm — APPROVED** [review]
neat

File: js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          match: [value],
+        })),
+      [allAssetOwners],
+    ),
+    menuWidth: '300px',
+    renderLabel: ({value}) => (
+      <Box flex={{direction: 'row', gap: 4, alignItems: 'center'}}>
+        <Icon name="account_circle" />
+        <TruncatedTextWithFullTextOnHover tooltipText={value} text={value} />
+      </Box>
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: You should use `UserDisplay` if it's a user imo

Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: Agreed. It's currently displaying strangely though -- I think we need to (1) unify OSS / Cloud user display styling and (2) Make a version that isn't a tag

image

What do you think about merging this version as is, and then switching to `UserDisplay` once the styling is updated?

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: Hmm, try AuditLogUserDisplay

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: I'm not opposed to making a new component for this similar to AuditLogUserDisplay (only in internal) for OSS

Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: The "claire" owner is the audit log user display, it also has the highlight on hover

If the owner doesn't exist as a cloud user, it displays using the oss user display. This is the "dataeng" owner in the picture

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: I see, we should change the rendering logic to only use UserDisplay for users and then have something separate for non users

**Comment by salazarm** [comment]
Can we also make sure this column gets sorted here https://github.com/dagster-io/internal/blob/60a57be197712efd98b04de697ea19072740a414/dagster-cloud/js_modules/app-cloud/src/LaunchpadHooksProvider/StaticFilterSorter.tsx#L10

This logic sorts the current user to the top of the list (so under owners I would see marco@dagsterlabs.com at the top)

File: js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx (hunk 1 above)
Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: yes

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: I think if `isFilter` is true we should never use the tag version.

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx:34]: eg here:
https://github.com/dagster-io/dagster/blob/f7d2611bdd46664496741b05bed3ce208d4f7b7e/js_modules/dagster-ui/packages/ui-core/src/runs/RunsFilterInput.tsx#L470

**Comment by clairelin135** [comment]
@salazarm added the sorting logic here: https://github.com/dagster-io/internal/pull/8665

File: js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx (hunk 1 above)
Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetOwnerFilter.tsx]: sweet, thanks, updated

**Pull request merged**